agent_template.py

```diff
@@ -0,0 +1,60 @@
+'''
+Adapted from agent template given in assignment 2
+'''
+from MarsRoverDisc import MarsRoverDisc
+import numpy as np
+
+class Agent(object):
+	def	__init__(self, env=None, gamma=0.99, theta = 0.00001, max_iterations=10000):
+		self.env = env
+		# Set of discrete actions for evaluator environment, shape - (|A|)
+		self.disc_actions = env.disc_actions
+		# Set of discrete states for evaluator environment, shape - (|S|)
+		self.disc_states = env.disc_states
+		# Set of probabilities for transition function for each action from every states, dicitonary of dist[s] = [s', prob, done, info]
+		self.Prob = env.Prob
+
+		self.gamma = gamma
+		self.theta = theta
+		self.max_iterations = max_iterations
+		self.value_policy, self.policy_function = None, None
+
+	def initialize(self):
+		self.value_policy, self.policy_function = self.solve()
+
+
+	def step(self, state):
+		action = self.policy_function[int(state)]
+		return action
+
+	def solve(self):
+		"""
+		insert solving mechanism here
+		"""
+		return
+
+
+def main():
+	myEnv = MarsRoverDisc(instance='0')
+	agent = Agent(env = myEnv)
+	agent.initialize()
+	state = myEnv.reset()
+	total_reward = 0
+
+	for step in range(myEnv.horizon):
+		action = agent.step(state)
+		next_state, reward, done, info = myEnv.step(action)
+		total_reward += reward
+		print()
+		print('step       = {}'.format(step))
+		print('state      = {}'.format(state), myEnv.disc_states[state])
+		print('action     = {}'.format(action), myEnv.disc_actions[action])
+		print('next state = {}'.format(next_state), myEnv.disc_states[next_state])
+		print('reward     = {}'.format(reward))
+		state = next_state
+		if done:
+			break
+	print("episode ended with reward {}".format(total_reward))
+	myEnv.close()
+
+main()
```

Comment:
Missing new line. Actually quite a few files are missing it but it's just a minor thing.